### PR TITLE
test: allow to run unit tests without installation

### LINKED ISSUE #1789: Not all unit test requiring install accept the RZ_PREFIX env var
Especially for distribution packaging, it is often useful to run all unit tests without installing, or with the install staged into a temporary directory.
The `RZ_PREFIX` env var can be used to tell the tests where the prefix is, but tests taking the prefix directly from `rz_sys_env()` will not respect that.
This makes patching like this necessary: https://paste.sr.ht/~thestr4ng3r/89fb2703bb74a8463d2d97d9c9cf240325ef706b

~~We could either:~~
* ~~Make `rz_sys_prefix()` also read the environment (maybe not so nice to have such logic in util/sys)~~
* ~~Add a unit-test specific function that returns `getenv("RZ_PREFIX") ? getenv("RZ_PREFIX") : rz_sys_prefix(NULL)`~~

Concrete plan to implement this, as discussed below:

- [ ] Add a function that generates the prefix by respecting `RZ_PREFIX` and others, as a heap-allocated local string, without setting any global state, in an appropriate place.
- [ ] Remove direct uses of rz_sys_prefix and similar functions (e.g. rz_str_rz_prefix) from all Rz modules
- [ ] Inject the prefix from the constructor, see https://github.com/rizinorg/rizin/issues/1789#issuecomment-933304576
- [ ] Change unit tests to make use of the function described above
- [ ] Move tests depending on rizin-testbins to integration-tests.

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Use build-level directories for types tests.

**Test plan**

```
$ podman  run -v $(pwd):/app -it fedora:35 bash
[root@b250f309fc6b app]# meson /tmp/build
[root@b250f309fc6b app]# meson test -C /tmp/build --suite unit
```
The above should work (after installing the right tools, meson, unit, gcc, etc.)

**Closing issues**

Closes #1789

### CODE REVIEW

File: test/unit/test_config.h.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// SPDX-FileCopyrightText: 2021 RizinOrg <info@rizin.re>
```
Comment (XVilka) [inline test/unit/test_config.h.in:1]: 2022

**Review by XVilka — CHANGES_REQUESTED** [review]
Please use `dist-` branch.

**Comment by XVilka** [comment]
It also broke Windows:
```
XX] Fail at line 609: struct array ptr func ptr type is ugly: expected struct alb { const char *b; int * const *a[][][][9]; wchar_t (*funk)(int a, const char *b); time_t t; };, got struct alb { const char *b; int * const *a[][][][9]; unknown_t (*funk)(int a, const char *b); unknown_t t; };.
test_enum_types OK
test_const_types OK
test_array_types OK
test_struct_func_types ERR
[XX] Fail at line 763: type as string with fcn ptr string: expected struct bla { int a; wchar_t (*func)(int a, const char *b); }, got struct bla { int a; unknown_t (*func)(int a, const char *b); }.
test_struct_array_types OK
test_struct_identifier_without_specifier OK
test_union_identifier_without_specifier OK
test_edit_types ERR
[XX] Fail at line 963: rz_type_declaration_as_string (new): expected struct alb { wchar_t * const b; int ***a[8][8][8]; float c; }, got struct alb { unknown_t * const b; int ***a[8][8][8]; float c; }.
test_references OK
test_addr_bits OK
stderr:
ERROR: Atomic type "int"
------------------------------------------------------------------------------
```
